src/scala/microsoft-spark-2-3/src/test/scala/org/apache/spark/util/dotnet/UtilsTest.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the .NET Foundation under one or more agreements.
+ * The .NET Foundation licenses this file to you under the MIT license.
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.apache.spark.util.dotnet
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.config.dotnet.Dotnet.DOTNET_IGNORE_SPARK_PATCH_VERSION_CHECK
+import org.junit.Assert.assertThrows
+import org.junit.Test
+import org.junit.function.ThrowingRunnable
+
+@Test
+class UtilsTest {
+
+  @Test
+  def shouldIgnorePatchVersion(): Unit = {
+    val conf = new SparkConf()
+    conf.set(DOTNET_IGNORE_SPARK_PATCH_VERSION_CHECK, true)
+
+    val sparkVersion = "2.3.5"
+    val sparkMajorMinorVersionPrefix = "2.3."
+    val supportedSparkVersions = Set[String]("2.3.0", "2.3.1", "2.3.2", "2.3.3", "2.3.4")
+
+    Utils.validateSparkVersions(
+      conf,
+      sparkVersion,
+      Utils.normalizeSparkVersion(sparkVersion),
+      sparkMajorMinorVersionPrefix,
+      supportedSparkVersions)
+  }
+
+  @Test
+  def shouldThrowForUnsupportedVersion(): Unit = {
+    val conf = new SparkConf()
+    conf.set(DOTNET_IGNORE_SPARK_PATCH_VERSION_CHECK, false)
+
+    val sparkVersion = "2.3.5"
+    val sparkMajorMinorVersionPrefix = "2.3."
+    val supportedSparkVersions = Set[String]("2.3.0", "2.3.1", "2.3.2", "2.3.3", "2.3.4")
+
+    assertThrows(
+      classOf[IllegalArgumentException],
+      new ThrowingRunnable {
+        override def run(): Unit = {
+          Utils.validateSparkVersions(
+            conf,
+            sparkVersion,
+            Utils.normalizeSparkVersion(sparkVersion),
+            sparkMajorMinorVersionPrefix,
+            supportedSparkVersions)
+        }
+      })
+  }
+
+  @Test
+  def shouldThrowForUnsupportedMajorMinorVersion(): Unit = {
+    val conf = new SparkConf()
+    conf.set(DOTNET_IGNORE_SPARK_PATCH_VERSION_CHECK, false)
+
+    val sparkVersion = "2.4.4"
+    val sparkMajorMinorVersionPrefix = "2.3."
+    val supportedSparkVersions = Set[String]("2.3.0", "2.3.1", "2.3.2", "2.3.3", "2.3.4")
+
+    assertThrows(
+      classOf[IllegalArgumentException],
```

Comment:
Can you use `intercept` and check the message as well (since we have two different places where we throw exceptions)? https://github.com/microsoft/hyperspace/blob/053b6b1e00f2b6dd62889e40d417dff1939079fc/src/test/scala/com/microsoft/hyperspace/index/CreateIndexTest.scala#L76

Reply:
Or we don't have `scalatest` set up?

Reply:
If `scalatest` is not possible, can you grep the exception message and make sure it's thrown at a different place?

Reply:
We don't have scalatest set up, only junit.  But we can still check exception message with it.